Framework/Core/include/Framework/AnalysisTask.h

```diff
@@ -643,9 +661,11 @@ DataProcessorSpec adaptAnalysisTask(ConfigContext const& ctx, Args&&... args)
   /// make sure options and configurables are set before expression infos are created
   std::apply([&options, &hash](auto&... x) { return (OptionManager<std::decay_t<decltype(x)>>::appendOption(options, x), ...); }, tupledTask);
 
+  auto processTuple = std::make_tuple(&T::process);
```

Comment:
Yes, so basically all we need now is:

```
if constexpr (has_multipleProcessDeclaration<T>) {
   processTuple = T::multipleProcessDeclaration();
} else {
   processTuple = std::make_tuple(&T::process);
}
```

Reply:
actually we can simply pass the processTuple as an argument, defaulting to `{&T::process}` and (almost) be done with the multiple process declaration...